### ISSUE TITLE
gpu-next: `glsl-shader-opts` values before shader parameters registered doesn't work.

### ISSUE THREAD

**Issue by natural-harmonia-gropius** [mpv-player/mpv#12039]
### Important Information

Provide following Information:
- mpv version https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20230727
- Platform and Version Windows 11
- Source of the mpv binary https://github.com/shinchiro/mpv-winbuild-cmake/releases

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

Use https://github.com/natural-harmonia-gropius/hdr-toys.
Add `glsl-shader-opts=L_sdr=100` to mpv.conf, before those profiles.

### Expected behavior

`L_sdr` set to 100

![image](https://github.com/mpv-player/mpv/assets/50797982/eb2835c5-f9a9-421f-b98b-303989f41131)

### Actual behavior

`L_sdr` keeps the default value (203).

![image](https://github.com/mpv-player/mpv/assets/50797982/fadeb55e-efda-4492-9c22-ed9a90fb0279)

### Log file

[mpv.log](https://github.com/mpv-player/mpv/files/12194943/mpv.log)

### Sample files

https://github.com/hooke007/MPV_lazy/assets/111181772/8dde1520-8ac6-431e-86a6-4dc1ffb487e2

**Comment by haasn** [comment]
From your log it seems you keep overriding `glsl-shader-opts` with a new value.

You probably want `glsl-shader-opts-append`.

**Comment by natural-harmonia-gropius** [comment]
> You probably want `glsl-shader-opts-append`.

Doesn't work too.

> From your log it seems you keep overriding glsl-shader-opts with a new value.

I thought `glsl-shader-opts` was something like `glsl-shader`...

I exec `set glsl-shader-opts sigma=1.0` via console and `set glsl-shader-opts L_sdr=100` via input.conf. both of them worked.

![image](https://github.com/mpv-player/mpv/assets/50797982/3c20a070-df19-44ed-b8cf-eadd01833f85)

**Comment by haasn** [comment]
> I thought glsl-shader-opts was something like glsl-shader...

No.

> Doesn't work too.

In input commands you need to use another command to add to list, I forget what.

**Comment by natural-harmonia-gropius** [comment]
> No. & In input commands you need to use another command to add to list, I forget what.

On my side, it actually works like `glsl-shader`, I can't feel any differences between `glsl-shader-opts` and `glsl-shader-opts-append`.

---

Back to the title, this is what I got.

before profile (shaders append)
![image](https://github.com/mpv-player/mpv/assets/50797982/37227d6e-c684-4a17-a17d-01b84294fec2)
![image](https://github.com/mpv-player/mpv/assets/50797982/2550ef77-9e08-4538-8513-77e4b81fcfce)

after profile (shaders append)
![image](https://github.com/mpv-player/mpv/assets/50797982/047242ce-ce3d-44ef-a2b0-61bfadfb0432)
![image](https://github.com/mpv-player/mpv/assets/50797982/c8d7e29f-1e77-4e7a-98d7-22f75dd488fa)

**Comment by natural-harmonia-gropius** [comment]
I did a quick test, `set glsl-shader-opts example=A` doesn't throw an error. I realized values before shader parameters registered can be invalid, so I think this issue can be close cause we can't know what will happens.

The source of this issue is I want to ensure glsl-shader-opts is applied to shaders. Shaders are append via auto-profile, sometimes append slower than 0.1 seconds causing glsl-shader-opts missed. I use `mp.add_timeout(0.1, fn)` here, but it doesn't always work, observe which property can handle this well?

https://github.com/natural-harmonia-gropius/mpv-config/blob/master/portable_config/scripts/hdr-toys-helper.lua

```lua
mp.observe_property("video-out-params", "native", function(_, value)
    if not value then return end

    local sig_peak = value["sig-peak"];
    if not sig_peak or sig_peak == 1 then
        set_L_hdr(1000)
        return
    end

    mp.add_timeout(0.1, function()
        set_L_hdr(sig_peak * 203)
    end)
end)
```
